src/WebJobs.Script.WebHost/Security/DefaultSecretsRepositoryFactory.cs

```diff
@@ -17,7 +17,13 @@ public ISecretsRepository Create(ScriptSettingsManager settingsManager, WebHostS
         {
             string secretStorageType = settingsManager.GetSetting(EnvironmentSettingNames.AzureWebJobsSecretStorageType);
             string storageString = AmbientConnectionStringProvider.Instance.GetConnectionString(ConnectionStringNames.Storage);
-            if (secretStorageType != null && secretStorageType.Equals("Blob", StringComparison.OrdinalIgnoreCase) && storageString != null)
+            string secretStorageSas = settingsManager.GetSetting(EnvironmentSettingNames.AzureWebJobsSecretStorageSas);
```

Comment:
I believe this change should preserve the current behavior if SecretStorageSas isn't provided.